udev_monitor.c

```diff
@@ -116,25 +110,21 @@ static void *udev_monitor_handle_event(void *ptr)
 {
     struct udev_monitor *udev_monitor = ptr;
     struct inotify_event *event;
-    struct epoll_event epoll[2];
+    struct pollfd poll_fd;
     char data[4096];
-    sigset_t set;
     ssize_t len;
     int i;
 
-    sigfillset(&set);
-    pthread_sigmask(SIG_BLOCK, &set, NULL);
-
-    while (epoll_wait(udev_monitor->efd, epoll, 2, -1) != -1) {
-        for (i = 0; i < 2; i++) {
-            if (epoll[i].data.fd != udev_monitor->pfd[0]) {
-                continue;
-            }
+    poll_fd.fd = udev_monitor->ifd;
+    poll_fd.events = POLLIN;
 
-            read(udev_monitor->pfd[0], data, 1);
-            write(udev_monitor->pfd[1], "1", 1);
-            pthread_barrier_wait(&udev_monitor->barrier);
-            return NULL;
+    while (1) {
+        if (poll(&poll_fd, 1, -1) == -1) {
```

Comment:
Check EINTR here. Kernel actually may fail to allocate resources for poll, so we can't simply "continue".